### DISCUSSION TITLE
Get username,password or hash from req.url

### DISCUSSION THREAD

**Discussion by rawkakani** [denoland/deno#17233] [Q&A] [answered]
Hi I have been trying to make a request to my deno server like this 

`https://username:password@locahost:9000/#some` and attempting to pull the username, password or the hash from the request body like [so](https://developer.mozilla.org/en-US/docs/Web/API/URL/username) `const {username, hash} = new URL(req.url)` 

`req:Request` coming from the handler of `serveTLS` from
https://deno.land/std@0.170.0/http/server.ts;

and I am failing to pick up the username, password or hash, has anyone been able to successfully get this right if so how and I can please be directed to the correct place

**Comment by ayame113** [comment, marked as answer]
#### hash

The URL hash value is not sent to the server. Therefore, it is not possible to retrieve the hash value on the server side.
(ref: https://stackoverflow.com/a/317769)

#### username and password

Username and password information is in the `authorization` header. This value is the Base64 encoding of ID and password joined by a single colon.

```ts
import { serve } from "https://deno.land/std@0.170.0/http/mod.ts";

serve((req) => {
  console.log(req.url);
  console.log(req.headers.get("authorization")); // => "Basic dXNlcjpwYXNz"
  return Response.json({ ok: true });
});
```

```shell
> curl http://user:pass@localhost:8000/
```

Perhaps this code will be helpful.↓
https://deno.land/x/basic_auth@v1.1.1/mod.ts?source#L36

**Reply by rawkakani** [reply]
@ayame113 thank you I was afraid that was the case
